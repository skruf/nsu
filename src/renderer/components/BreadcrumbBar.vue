<style lang="stylus">
.breadcrumb-bar
  margin-bottom 15px
  width 100%
  display flex
  align-items center
  justify-content space-between
  .el-button
    padding 0
    & + .el-button
      margin-left 5px
.el-breadcrumb
  font-size 13px
  padding-left 1px
  padding-right 1px
.el-breadcrumb__inner
  text-transform capitalize
  &:hover
    color var(--primary-color)
    cursor pointer
.el-breadcrumb__separator
  color var(--text-color)
</style>

<template>
  <div class="breadcrumb-bar">
    <el-breadcrumb separator-class="el-icon-arrow-right">
      <el-breadcrumb-item
        v-for="(item, index) in paths"
        :key="index"
        :to="{ path: item.to }"
      >
        {{ item.label }}
      </el-breadcrumb-item>
    </el-breadcrumb>

    <div>
      <el-button
        type="text"
        icon="el-icon-arrow-left"
        @click="navBack"
      />
      <el-button
        type="text"
        icon="el-icon-arrow-right"
        @click="navForward"
      />
    </div>
  </div>
</template>

<script>
import { Breadcrumb, BreadcrumbItem } from "element-ui"

export default {
  name: "BreadcrumbBar",

  components: {
    ElBreadcrumb: Breadcrumb,
    ElBreadcrumbItem: BreadcrumbItem
  },

  props: {
    paths: { type: Array, required: true }
  },

  methods: {
    navBack() {
      this.$router.go(-1)
    },
    navForward() {
      this.$router.go(1)
    }
  }
}
</script>
