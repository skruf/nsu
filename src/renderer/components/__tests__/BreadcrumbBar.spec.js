import createLocalVue from "@/utils/tests/createLocalVue"
import { Breadcrumb, BreadcrumbItem, Button } from "element-ui"
import BreadcrumbBar from "@/components/BreadcrumbBar"
import mount from "@/utils/tests/mount"

const localVue = createLocalVue([ Breadcrumb, BreadcrumbItem, Button ])

describe("BreadcrumbBar", () => {
  const build = (overrides = {}, deep = false) => {
    return mount(BreadcrumbBar, localVue, overrides, deep)
  }

  it("should render the component", () => {
    const wrapper = build({ propsData: { paths: [] } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have a breadcrumb with breadcrumb items", () => {
    const wrapper = build({
      propsData: {
        paths: [
          { to: "/ranges", label: "Ranges" },
          { to: "/ranges/all", label: "All" }
        ]
      }
    })
    expect(wrapper.find(Breadcrumb).exists()).toBe(true)
    expect(wrapper.findAll(BreadcrumbItem)).toHaveLength(2)
  })

  it("should have functioning prev / next buttons", () => {
    const wrapper = build({
      propsData: {
        paths: [
          { to: "/ranges", label: "Ranges" },
          { to: "/ranges/all", label: "All" }
        ]
      },
      mocks: {
        $router: { go: jest.fn() }
      }
    }, true)

    const buttons = wrapper.findAll(Button)

    buttons.at(0).trigger("click")
    buttons.at(1).trigger("click")

    expect(buttons).toHaveLength(2)

    expect(wrapper.vm.$router.go).toHaveBeenNthCalledWith(1, -1)
    expect(wrapper.vm.$router.go).toHaveBeenNthCalledWith(1, -1)
  })
})
