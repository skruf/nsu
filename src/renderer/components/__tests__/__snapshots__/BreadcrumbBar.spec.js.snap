// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbBar should render the component 1`] = `
<div class="breadcrumb-bar">
  <el-breadcrumb-stub separator="/" separatorclass="el-icon-arrow-right"></el-breadcrumb-stub>
  <div>
    <el-button-stub type="text" icon="el-icon-arrow-left" nativetype="button"></el-button-stub>
    <el-button-stub type="text" icon="el-icon-arrow-right" nativetype="button"></el-button-stub>
  </div>
</div>
`;
